pkg/parse/run_test.go

```diff
@@ -55,7 +56,7 @@ const (
 	symLink = "rev"
 )
 
-func newParser(t *testing.T, fs FileSource, renderingEnabled bool) Parser {
+func newParser(t *testing.T, fs *FileSource, renderingEnabled bool, retryPeriod time.Duration, pollingPeriod time.Duration) Parser {
```

Comment:
Passing nil to a constructor is a code smell. It makes it harder to read the constructor invocation without looking at the constructor implementation. 

Similar to the DefaultRunOpts, I would suggest using an EmptyFiles or FakeFiles or some other such Files impl or constructor for the case you're adding, with a well defined behavior, instead of adding conditionals into the constructor.

If you absolutely need optional arguments in a constructor, it's usually preferable to have multiple clearly named constructors. But I don't think that's the case here, since it's just a test that doesn't get to a branch that needs the Files.

Reply:
Done. I added a function in `run_test.go` that creates an empty FileSource